[libsndfile] Update to 1.0.29

evpobr opened the pull request:
Updates:

* Bump to official 1.0.29 release
* Two Git patches removed as merged to upstream
* Request Python3 when building shared library to generate export symbol files
* `CONTROL` replaced by `vcpkg.json`
* Removed `BUILD_EXECUTABLES` code. It was always `OFF` anyway
* Use new options instead of deprecated ones

Important note: target name is changed to`SndFile::sndfile` from upstream.

evpobr committed:
[libsndfile] Update to 1.0.29

evpobr commented:
Don't understand why Python is not found.

evpobr commented:
Thanks @LilyWangL :smile:

evpobr commented:
@LilyWangL , this PR is ready.

BillyONeal commented on ports/libsndfile/portfile.cmake:
```diff
@@ -1,26 +1,22 @@
 
 if("external-libs" IN_LIST FEATURES)
-    set(SNDFILE_WITHOUT_EXTERNAL_LIBS OFF)
+    set(SNDFILE_WITH_EXTERNAL_LIBS ON)
 else()
-    set(SNDFILE_WITHOUT_EXTERNAL_LIBS ON)
+    set(SNDFILE_WITH_EXTERNAL_LIBS OFF)
 endif()
```
It seems like this should do:

vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES external-libs SNDFILE_WITH_EXTERNAL_LIBS);

(and then below for `vcpkg_configure_cmake` do `${FEATURE_OPTIONS}`)

BillyONeal committed:
Restore port-version and bump to 11.

BillyONeal committed:
Use vcpkg_check_features.

I checked for "-DENABLE_EXTERNAL_LIBS=ON" and "-DENABLE_EXTERNAL_LIBS=ON" according to the feature setting.

evpobr reviewed (commented):
@BillyONeal , it is final 1.0.29 release.

BillyONeal commented:
@evpobr: Yes, but the old one was marked 1.0.29-10 and we want the version number to be increasing over time (assuming the upstream is going forward over time)

BillyONeal commented:
(Note that our `version-portversion` convention in the CONTROL files was *not* semver)

BillyONeal commented:
Thanks for your contribution!
